doc/api/packages.md

```diff
@@ -83,6 +85,41 @@ future-proof the package in case the default type of Node.js ever changes, and
 it will also make things easier for build tools and loaders to determine how the
 files in the package should be interpreted.
 
+### Modules loaders
+
+Node.js has two system for resolving a specifier and load modules.
+
+There is the CommonJS module loader:
+
+* It is fully synchronous.
+* It is monkey patchable.
+* When resolving a specifier, if no exact match is found, it will try to add
+  extensions (`.js`, `.json`, and finally `.node`).
+* It supports [folders as modules][].
+* It treats `.json` as JSON text files.
+* `.node` files are interpreted as compiled addon modules loaded with
+  `process.dlopen()`.
+* It treats files that do not have `.json` or `.node` extension as JavaScript
+  text files.
+* It cannot be used to load ECMAScript modules. Attempting to do so will result
+  in a [`ERR_REQUIRE_ESM`][] error.
+* It can be accessed using `require` function.
+
+There is the ECMAScript module loader:
+
+* It is Asynchronous.
+* It is not monkey patchable, can be customized using [loader hooks][].
+* No extension searching, the specifier must point to the exact URL of the file.
```

Comment:
```suggestion
* It is asynchronous.
* It is not monkey patchable, but can be customized using [loader hooks][].
* Specifiers referencing files must include file extensions; Node.js will not
  search for them.
```

Reply:
must include file extensions or match the "exports" field of the package being imported?

Reply:
I agree with Jordan, it's not correct to say specifier must include file extensions. Maybe we can reuse the wording from https://github.com/nodejs/node/blob/1b1159ae502a134c48427f6351627d32767539af/doc/api/esm.md#L149-L151

Reply:
directory indexes must only be fully specified *when using relative or absolute specifiers* tho, which i'm not sure is clear enough without being part of the same sentence, or stating it twice.

Reply:
I went with https://github.com/nodejs/node/blob/2cea6a8fcd4ba7ccb9a7dfe632ba33b489786f42/doc/api/packages.md#L113-L114 let me know if you think it needs more work.

Reply:
I’m fine with the terms, but that particular sentence has a comma splice, which is incorrect grammar.

```markdown
 * It does no extension searching. A file extension must be provided
   when the specifier is a relative or absolute file URL.
```